src/solver/Row.java

```diff
@@ -0,0 +1,42 @@
+/**
+Я решительно не понимаю 1 важную вещь. Сейчас в конструкторе у меня цикл и вроде как лишние приравнения, сделано это потому,
+что иначе при задании массива строк у нас присваивание будет (как я понимаю) по ссылке и в такой ситуации:
+Row[] row = new Row[5]; 
+row[0] = first_row; ... row[4] = fifth_row; в row[0] будет храниться  fifth_row и я не совсем понимаю почему. Точнее, я не понимаю
+почему при таком задании конструктора как тут, все работает правильно.
+*/
+class Row { // N на 1 больше, чем вход, т.к. тут я учитываю и ответ (то, что справа от равно)
+	int N;
+	ComplexNumber[] line = new ComplexNumber[1];
```

Comment:
You don't need to write an initial value for the array if you will rewrite it anyway. Like in the constructor.